tests/api2/assets/REST/pool.py

```diff
@@ -0,0 +1,24 @@
+import contextlib
+
+from functions import DELETE, GET, POST, PUT, wait_on_job
+
+
+@contextlib.contextmanager
+def dataset(pool_name, dataset_name):
```

Comment:
I think from standpoint of common test assets, it would be useful to allow specifying dataset properties as `options` kwarg and acl or mode as kwarg.

C.f.
https://github.com/truenas/middleware/blob/019654f9895901e48f02c528cca5fef66a32157d/tests/api2/test_300_nfs.py#L80